### PR TITLE
Fixes header sig verification

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
## Reasoning behind the pull request
- The proof verification (aggregated signature) with equivalent proofs is for the unaltered header, so it also needs to be verified on the unaltered header
- Add more safety and integrity checks
  
## Proposed changes
- Leader signature is not removed from the header before verifying the proof aggregated signature
- Make sure new header fields (proof related) are validated for both before activation - throw an error if the new fields are present, and after activation - throw an error if the fields are not present

## Testing procedure
- Unit tests
- Regular system test with transitioning from old to new consensus will be done on the feat branch
   
## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: integrationTests/consensus/consensusSigning_test.go [hunk 1]
```diff
@@ -69,6 +71,8 @@ func TestConsensusWithInvalidSigners(t *testing.T) {
 		t.Skip("this is not a short test")
 	}
 
+	_ = logger.SetLogLevel("*:DEBUG")
```
Comment (sstanculeanu) [inline integrationTests/consensus/consensusSigning_test.go:74]: is this still needed?

Comment (AdoAdoAdo) [reply integrationTests/consensus/consensusSigning_test.go:74]: no, forgot this

File: consensus/spos/bls/v1/subroundBlock.go [hunk 2]
```diff
@@ -535,6 +545,13 @@ func (sr *subroundBlock) receivedBlockHeader(ctx context.Context, cnsDta *consen
 	return blockProcessedWithSuccess
 }
 
+func headerHasProof(headerHandler data.HeaderHandler) bool {
+	if check.IfNil(headerHandler) {
+		return true
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply consensus/spos/bls/v1/subroundBlock.go:550]: meant to return false

File: consensus/spos/bls/v1/subroundBlock.go [hunk 3]
```diff
@@ -535,6 +545,13 @@ func (sr *subroundBlock) receivedBlockHeader(ctx context.Context, cnsDta *consen
...
+	}
+	return headerHandler.GetPreviousProof() != nil
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply consensus/spos/bls/v1/subroundBlock.go:552]: right, forgot the Get was returning an interface not a pointer to the proof

File: integrationTests/consensus/consensusSigning_test.go [hunk 4]
```diff
@@ -103,7 +107,7 @@ func TestConsensusWithInvalidSigners(t *testing.T) {
 		go checkBlockProposedEveryRound(numCommBlock, nonceForRoundMap, mutex, chDone, t)
 
 		extraTime := uint64(2)
-		endTime := time.Duration(roundTime)*time.Duration(numCommBlock+extraTime)*time.Millisecond + time.Minute
+		endTime := time.Duration(roundTime)*time.Duration(numCommBlock+extraTime)*time.Millisecond + 10*time.Minute
```
Comment (ssd04) [inline integrationTests/consensus/consensusSigning_test.go:110]: do we want to extend end time so much?

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**